src/animate/AnimatorTimeline.js

```diff
@@ -37,21 +40,29 @@ class AnimatorTimeline {
      * @private
      */
     destroy() {
-        this.init(null, null, false, 0, 0, null);
+        this.instance.beforeUpdateTimeline = null;
+        this.init(null, 0, 0, false, null);
         AnimatorTimeline._pool.push(this);
     }
 
     /**
      * Is the animation complete
-     * @method update
+     * @method before
      * @return {Boolean} 
      * @private
      */
-    update() {
-        if (this.instance.currentFrame >= this.end) {
+    update(instance) {
+        if (instance.currentFrame >= this.end) {
+
+            // In case we over-shoot the current frame becuase of low FPS
+            instance.currentFrame = this.end;
+
             if (this.loop) {
-                this.instance.gotoAndPlay(this.start);
+                instance.gotoAndPlay(this.start);
```

Comment:
The action running code in updateTimeline() recognizes when you loop around, and when it does that it does all the actions from the previous frame until the end of the timeline, and then from the beginning of the timeline to the new frame. That is great for self playing movieclips, but not for Animator controlled ones that don't play the entire timeline. You could set the current frame to the end, call updateTimeline() on the movieclip immediately, and then call gotoAndPlay(), or you could manually run just the actions (or add a function for doing so).